plugins/node/opentelemetry-instrumentation-memcached/src/instrumentation.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import {
+  isWrapped,
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+} from '@opentelemetry/instrumentation';
+import type * as Memcached from 'memcached';
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
+import * as utils from './utils';
+import { InstrumentationConfig } from './types';
+import { VERSION } from './version';
+
+export class Instrumentation extends InstrumentationBase<typeof Memcached> {
+  static readonly COMPONENT = 'memcached';
+  static readonly COMMON_ATTRIBUTES = {
+    [SemanticAttributes.DB_SYSTEM]: Instrumentation.COMPONENT,
+  };
+  static readonly DEFAULT_CONFIG: InstrumentationConfig = {
+    collectCommand: false,
+  };
+
+  constructor(config: InstrumentationConfig = Instrumentation.DEFAULT_CONFIG) {
+    super(
+      '@opentelemetry/instrumentation-memcached',
+      VERSION,
+      Object.assign({}, Instrumentation.DEFAULT_CONFIG, config)
+    );
+  }
+
+  setConfig(config: InstrumentationConfig = Instrumentation.DEFAULT_CONFIG) {
```

Comment:
I'm not sure to understand why you reset the default config here ? It doesn't make sense to set the config without actualling give it a config right ?

Reply:
For resetting it to defaults, it does. I don't mind whichever way. Do you think it should be changed?

Reply:
I don't see why an user would want to reset to the default config at runtime (althrough its handy in tests). I would prefer to match the signature of the implementation in the `BaseInstrumentation` class

Reply:
OK! Changed it.